modules/private/class-jetpack-private.php

```diff
@@ -0,0 +1,329 @@
+		remove_filter( 'bloginfo', array( __CLASS__, 'privatize_blog_mask_blog_name' ) );
+	}
+
+	/**
+	 * Filters new comments so that users can't comment on private blogs
+	 *
+	 * @param array $comment Documented in wp-includes/comment.php.
+	 */
+	public static function privatize_blog_comments( $comment ) {
+		self::privatize_blog( null );
+		return $comment;
+	}
+
+	/**
+	 * Extend the 'Site Visibility' privacy options to also include a private option
+	 **/
+	public static function privatize_blog_priv_selector() {
+		?>
+		<style>
+			.jetpack-private__setting-disabled {
+				font-weight: bold;
+				padding: 10px;
+			}
+			.option-site-visibility fieldset {
+				display: none;
+			}
+		</style>
+		</fieldset>
+		<div class="jetpack-private__setting-disabled highlight">
+			<?php
+			/* translators: URL for Jetpack dashboard. */
+			wp_kses( sprintf( __( 'This setting is ignored because you <a href="%s">made your site private</a>', 'jetpack' ), admin_url( 'admin.php?page=jetpack' ) . '#/security?term=private' ), array( 'a' => array( 'href' => true ) ) );
+			?>
+		</div>
+		<fieldset>
+		<?php
+	}
+
+	/**
+	 * Don't let search engines index private sites
+	 * or sites not deemed publicly available, like deleted, archived, spam.
+	 *
+	 * @param string $output Robots.txt output.
+	 */
+	public static function private_robots_txt( $output ) {
+		$output  = "User-agent: *\n"; // Purposefully overriding current output; we only want these rules.
+		$output .= "Disallow: /\n";
+		return $output;
+	}
+
+	/**
+	 * Output the meta tag that tells Pinterest not to allow users to pin
+	 * content from this page.
+	 * https://support.pinterest.com/entries/21063792-what-if-i-don-t-want-images-from-my-site-to-be-pinned
+	 */
+	public static function private_no_pinning() {
+		echo '<meta name="pinterest" content="nopin" />';
+	}
+
+	/**
+	 * Prevents ajax and post requests on private blogs for users who don't have permissions
+	 */
+	public static function private_blog_prevent_requests() {
+		global $pagenow, $wpdb;
+
+		$this_current_user     = wp_get_current_user();
+		$is_ajax_request       = defined( 'DOING_AJAX' ) && DOING_AJAX;
+		$is_admin_post_request = ( 'admin-post.php' === $pagenow );
+
+		if ( is_super_admin() ) {
+			return;
+		}
+
+		// Make sure we are in the right code path, if not bail now.
+		if ( ! is_admin() || ( ! $is_ajax_request && ! $is_admin_post_request ) ) {
+			return;
+		}
+
+		if ( ! self::is_private_blog_user( $wpdb->blogid, $this_current_user ) ) {
+			wp_die( '', 403 );
+		}
+	}
+
+	/**
+	 * Prevents ajax requests on private blogs for users who don't have permissions
+	 *
+	 * @param string    $action The Ajax nonce action.
+	 * @param false|int $result The result of the nonce check.
+	 */
+	public static function private_blog_ajax_nonce_check( $action, $result ) {
+		global $wpdb;
+
+		$this_current_user = wp_get_current_user();
+
+		if ( is_super_admin() ) {
```

Comment:
Same here.

Reply:
Done in [2a1e665](https://github.com/Automattic/jetpack/pull/10333/commits/2a1e66582a8bcc4cde03e35e1abf46b06a75d8de)